### PR TITLE
Refactor modal

### PR DESCRIPTION
**Pull request by barmac**
This refactoring decouples modals from the `App#setModal` which allows to use them via plugins in the future. Also, thanks to usage of `ReactDOM#createPortal`, the modal's style is independent from the parent component style. This allows to display modal even if the parent component is hidden via `display: none`.

### CODE REVIEW

File: client/src/app/primitives/modal/Modal.js [hunk 1]
```diff
@@ -9,30 +9,40 @@
...
 
 import classNames from 'classnames';
 
-import css from './ModalWrapper.less';
+import css from './Modal.less';
 
 
-const ModalWrapper = props => {
-  const handleBackgroundClick = event => {
+function Modal(props) {
```
Comment (nikku) [inline client/src/app/primitives/modal/Modal.js]: Is this the right moment to introduce React hooks into our code base? I'm not sure about this one.

Comment (barmac) [reply client/src/app/primitives/modal/Modal.js]: We've bumped React version to 16.8 about 4 months ago and it seems the hooks are not going anywhere. Why should we bother?

Comment (barmac) [reply client/src/app/primitives/modal/Modal.js]: Anyway, the class alternative is not overly verbose or anything. Let's leave hooks out of this PR.

Comment (barmac) [reply client/src/app/primitives/modal/Modal.js]: And yes, the `useRef` usage looked a bit clumsy, so it's not only about not hurting but the actual benefit of removing hooks out of this ;)

**Pull request merged**